tests/integration/test_dbapi_integration.py

```diff
@@ -123,22 +124,114 @@ def test_string_query_param(trino_connection):
     assert rows[0][0] == "six'"
 
 
+def test_decimal_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    cur.execute("SELECT ?", params=(Decimal('0.142857'),))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == Decimal('0.142857')
+
+
 def test_datetime_query_param(trino_connection):
     cur = trino_connection.cursor()
 
-    cur.execute("SELECT ?", params=(datetime(2020, 1, 1, 0, 0, 0),))
+    params = datetime(2020, 1, 1, 16, 43, 22, 320000)
+
+    cur.execute("SELECT ?", params=(params,))
     rows = cur.fetchall()
 
-    assert rows[0][0] == "2020-01-01 00:00:00.000"
+    assert rows[0][0] == params
+    assert cur.description[0][1] == "timestamp"
+
+
+def test_datetime_with_time_zone_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    params = datetime(2020, 1, 1, 16, 43, 22, 320000, tzinfo=pytz.timezone('CET'))
 
     cur.execute("SELECT ?",
-                params=(datetime(2020, 1, 1, 0, 0, 0, tzinfo=pytz.utc),))
+                params=(params,))
     rows = cur.fetchall()
 
-    assert rows[0][0] == "2020-01-01 00:00:00.000 UTC"
+    assert rows[0][0] == params
     assert cur.description[0][1] == "timestamp with time zone"
 
 
+def test_datetime_with_time_zone_numeric_offset(trino_connection):
+    cur = trino_connection.cursor()
+
+    cur.execute("SELECT TIMESTAMP '2001-08-22 03:04:05.321 -08:00'")
+    rows = cur.fetchall()
+
+    assert rows[0][0] == datetime.strptime("2001-08-22 03:04:05.321 -08:00", "%Y-%m-%d %H:%M:%S.%f %z")
+
+
+def test_special_datetimes_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    for special_date in (
+            datetime.fromtimestamp(1603589478, pytz.timezone('Europe/Warsaw')),
+    ):
+        params = special_date
+
+        cur.execute("SELECT ?", params=(params,))
+        rows = cur.fetchall()
+
+        assert rows[0][0] == params
+
+
+def test_date_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    params = datetime(2020, 1, 1, 0, 0, 0).date()
+
+    cur.execute("SELECT ?", params=(params,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == params
+
+
+def test_special_dates_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    for params in (
+        # datetime(-1, 1, 1, 0, 0, 0).date(),
+        # datetime(0, 1, 1, 0, 0, 0).date(),
+        datetime(1752, 9, 4, 0, 0, 0).date(),
+        datetime(1970, 1, 1, 0, 0, 0).date(),
+    ):
+        cur.execute("SELECT ?", params=(params,))
+        rows = cur.fetchall()
+
+        assert rows[0][0] == params
+
+
+def test_time_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    params = time(12, 3, 44, 333000)
+
+    cur.execute("SELECT ?", params=(params,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == params
+
+
+@pytest.mark.skip(reason="time with time zone currently not supported")
```

Comment:
What's required for it to be supported?

Reply:
Instead of removing can you add a test that fails? It helps document and also helps ensure tests get updated if behaviour changes.

Reply:
@mdesmet Was this addressed? Can you point me to the method which verifies this? I think this is also the source of my confusion regarding https://github.com/trinodb/trino-python-client/pull/160#discussion_r821385194.

Maybe add a comment there too indicating that time with time zone is not supported.